### PR TITLE
OP-18616: added api to valid ldap Connection/Authentication.

### PR DESCRIPTION
**Pull request by rahul-chekuri**
https://devopsmx.atlassian.net/browse/OP-18675

UTC
With LDAP standard props in place was able to add LDAP to keycloak
Validated test connection with/without valid values
Validated test authentication with/without valid values

### CODE REVIEW

File: docker_build/gate.yml [hunk 1]
```diff
@@ -133,6 +133,27 @@ keycloak:
   serverUrl: http://localhost:8080/admin
   username: myuser
   password: password
+  ldapConfigProps:
```
Comment (rahul-chekuri) [inline docker_build/gate.yml:136]: These are some properties with defaults that will be common for any LDAP.

Comment (rahul-chekuri) [reply docker_build/gate.yml:136]: Without these we cant create LDAP in keycloak

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/controller/AuthConfigurerController.java [hunk 2]
```diff
@@ -114,4 +118,15 @@ public ResponseEntity<AuthProviderModel> disableAuthProvider(
     log.trace("disableAuthProvider ended");
     return response;
   }
+
+  @PostMapping(
+      value = "/authProvider/ldap/testLDAPConnection",
+      produces = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<String> validateLdapAuthConnection(
```
Comment (rahul-chekuri) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/controller/AuthConfigurerController.java:125]: New API to testLDAPConnection.

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/model/KeycloakProperties.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+package com.opsmx.spinnaker.gate.model;
+
+import java.util.Map;
+import lombok.Data;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+@Data
+@Component
+@ConfigurationProperties(prefix = "keycloak.ldapconfigprops")
```
Comment (Pranav-b-7) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/model/KeycloakProperties.java:26]: Add feature flag condition here and make sure the autowiring does not fail when the flag is disabled

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/controller/AuthConfigurerController.java [hunk 4]
```diff
@@ -114,4 +118,15 @@ public ResponseEntity<AuthProviderModel> disableAuthProvider(
...
+      @RequestBody TestLdapConnectionRepresentation testLdapConnectionRepresentation) {
+    log.trace("validateLdapAuthConnection started");
+    String status = keycloakAuthUtils.validate(testLdapConnectionRepresentation);
```
Comment (Pranav-b-7) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/controller/AuthConfigurerController.java:128]: Why the util class is called directly from controller ? it has to be from the service layer rit ?

**Review by Pranav-b-7 — CHANGES_REQUESTED** [review]

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/controller/AuthConfigurerController.java (hunk 4 above)
Comment (rahul-chekuri) [reply gate-web/src/main/java/com/opsmx/spinnaker/gate/controller/AuthConfigurerController.java:128]: this is only for LDAP not for all so calling directly

Comment (rahul-chekuri) [reply gate-web/src/main/java/com/opsmx/spinnaker/gate/controller/AuthConfigurerController.java:128]: service layer is generic one for all types

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/model/KeycloakProperties.java (hunk 3 above)
Comment (rahul-chekuri) [reply gate-web/src/main/java/com/opsmx/spinnaker/gate/model/KeycloakProperties.java:26]: Through the feature flag that we have at the controller, we can simply disable this feature. It need not be everywhere as this is the new code.

**Pull request merged**